sgt-back-end/index.js

```diff
@@ -0,0 +1,131 @@
+const express = require('express');
+const app = express();
+
+const pg = require('pg');
+const db = new pg.Pool({
+  connectionString: 'postgres://dev:dev@localhost/studentGradeTable',
+  ssl: {
+    rejectUnauthorized: false
+  }
+});
+
+app.get('/api/grades', (req, res) => {
+  const sql = `
+  select *
+  from "grades"
+  order by "gradeId"
+  `;
+  db.query(sql)
+    .then(result => {
+      const resultData = result.rows;
+      res.json(resultData);
+      res.status(200);
+    })
+    .catch(err => {
+      console.error(err);
+      res.status(500).json({
+        error: 'An unexpected error occcured'
+      });
+    });
+});
+
+app.use(express.json());
+
+app.post('/api/grades', (req, res) => {
+  const score = Number(req.body.score);
+  const text = `insert into "grades" ("name", "course", "score")
+                values ($1, $2, $3)
+                returning *`;
+  const values = [req.body.name, req.body.course, score];
+  if (!req.body.name || !req.body.course || !req.body.score || score < 0 || !Number.isInteger(score)) {
+    res.status(400).json({
+      error: 'invalid input'
+    });
+  } else {
+    db.query(text, values)
+      .then(result => {
+        const resultData = result.rows[0];
+        res.json(resultData);
+        res.status(200);
```

Comment:
For creating a new record, the client should receive a `201 Created` response. You'll need to reverse the order of these two lines, because if you do `res.json()` or `res.send()` before `res.status()`, then Express will not set the correct status code.

HTTP response messages have the status code before the body, so you have to set the custom status code before sending the body.